sdk/search/azure-search/src/main/java/com/azure/search/models/Skill.java

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.util.List;
+
+/**
+ * Abstract base class for skills.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type", defaultImpl = Skill.class)
+@JsonTypeName("Skill")
+@JsonSubTypes({
```

Comment:
Skills that are missing/we'll want to add:
DocumentExtrationSkill (already in production)
CustomEntityLookupSkill (in upcoming release)
PIIDetectionSkill (in upcoming release)

Also interesting that the deprecated skill NamedEntityRecognitionSkill isn't in this list, wonder why that is.

Reply:
@careyjmac When you add new skills, since they cross API versions, you'll need to add them to all versions of the Swagger spec. Something to keep in mind.

Not sure about the `JsonSubTypes` thing. Maybe a bug in AutoRest? @navalev -- Any idea why the list of sub-types is not complete?

Reply:
@brjohnstmsft There is a class model for NamedEntityRecognitionSkill and it does extend Skill model, so the JsonSubTypes annotication should include it. @jianghaolu can you assist?

Reply:
@brjohnstmsft I double checked it, and seems that in the swagger sepc NamedEntityRecognitionSkill is set to be excluded. It has the property **x-ms-external** set to true, which according to the Autrest docs excludes an object from code generation.  See https://github.com/Azure/autorest/blob/06de16e52c241e87e3447ee7a265a9b06886ef57/docs/extensions/readme.md

Reply:
@navalev That was probably because we had to manually add `[Obsolete]` to the class in the .NET SDK. I think we should remove `x-ms-external` from the Swagger and instead delete the generated file in our .NET build script. We'll take care of this when we do a pass over the Swagger. It shouldn't block this PR.